### PR TITLE
Crud admin

### PR DESCRIPTION
**Pull request by CesarCachay**
- Add controllers of resources
- Add views of resources
- Add associations

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,4 +1,16 @@
...
+  get 'bookings/new'
+  get 'bookings/edit'
+  get 'rooms/index'
+  get 'rooms/show'
+  get 'rooms/new'
+  get 'rooms/edit'
+  get 'hotels/index'
+  get 'hotels/show'
+  get 'hotels/new'
+  get 'hotels/edit'
```
Comment (cristianbgp) [inline config/routes.rb]: Please put all the resources inside a namespace called `admin`

File: app/models/booking.rb [hunk 2]
```diff
@@ -1,4 +1,5 @@
 class Booking < ApplicationRecord
   belongs_to :user
   belongs_to :room
+  belongs_to :hotel
```
Comment (stg101) [inline app/models/booking.rb]: Great association created for our booking system

File: app/models/hotel.rb [hunk 3]
```diff
@@ -1,2 +1,4 @@
 class Hotel < ApplicationRecord
+  has_many :rooms
+  has_many :promotions, as: :promotionable
```
Comment (stg101) [inline app/models/hotel.rb]: Good use of polymorphism

File: app/models/user.rb [hunk 4]
```diff
@@ -4,6 +4,7 @@ class User < ApplicationRecord
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :validatable,
          :omniauthable, omniauth_providers: %i[facebook github]
+  has_many :bookings
```
Comment (stg101) [inline app/models/user.rb]: Great, this allow to our users to have more than one booking

File: config/routes.rb [hunk 5]
```diff
@@ -1,4 +1,10 @@
 Rails.application.routes.draw do
   devise_for :users, controllers: { omniauth_callbacks: 'users/omniauth_callbacks' }
   root to: 'home#index'
+
+  namespace :admin do
```
Comment (stg101) [inline config/routes.rb]: Nice, it helps us to group our resources for admin routes.

**Pull request merged**